src/x86_64/Gtrace.c

```diff
@@ -430,7 +432,10 @@ tdep_trace (unw_cursor_t *cursor, void **buffer, int *size)
     d->stash_frames = 0;
     return -UNW_ENOMEM;
   }
+  if (cache->frame_rc > 0)
+    return -UNW_EINVAL;
 
+  cache->frame_rc = 1;
```

Comment:
Won't this fail if there are multiple nested requests? It is a 'reference count', but apparently it doesn't actually count?

Reply:
> Won't this fail if there are multiple nested requests? 

Yes

> It is a 'reference count', but apparently it doesn't actually count?

At first, I thought of using `frame_rc` to represent the reference count of `unw_trace_cache_t.frames`, and `munmap(frames)` should be delayed until when `frame_rc` is 0. Then I realized that the current PR is just a workaround, let’s not make it so complicated. Currently, `frame_rc` is used to indicate whether `frames` is being used, if so, nested requests will not be allowed to use `frames` again.

Reply:
I can rename `frame_rc` to`in_use` if needed.

Reply:
@djwatson